dkg/paranet.py

```diff
@@ -83,6 +85,318 @@ def create(
+
+        self._request_paranet_curated_node_access(
+            knowledge_asset_storage,
+            knowledge_asset_token_id,
+        )
+
+        return {
+            "paranetUAL": ual,
+            "paranetId": Web3.to_hex(
+                Web3.solidity_keccak(
+                    ["address", "uint256"],
+                    [knowledge_asset_storage, knowledge_asset_token_id],
+                )
+            ),
+            "operation": "Request to be included in curated paranet nodes"
+        }
+    
+    _approve_curated_node = Method(BlockchainRequest.approve_curated_node)
+    
+    def approve_curated_node(self, ual: UAL, identity_id: int) -> dict[str, str | HexStr | TxReceipt]:
+        parsed_ual = parse_ual(ual)
+        knowledge_asset_storage, knowledge_asset_token_id = (
+            parsed_ual["contract_address"],
+            parsed_ual["token_id"],
+        )
+
+        self._approve_curated_node(
+            knowledge_asset_storage,
+            knowledge_asset_token_id,
+            identity_id,
+        )
+
+        return {
+            "paranetUAL": ual,
+            "paranetId": Web3.to_hex(
+                Web3.solidity_keccak(
+                    ["address", "uint256"],
+                    [knowledge_asset_storage, knowledge_asset_token_id],
+                )
+            ),
+            "operation": "Approve node's curated paranet access request"
+        }
+    
+    _reject_curated_node = Method(BlockchainRequest.reject_curated_node)
+    
+    def reject_curated_node(self, ual: UAL, identity_id: int) -> dict[str, str | HexStr | TxReceipt]:
+        parsed_ual = parse_ual(ual)
+        knowledge_asset_storage, knowledge_asset_token_id = (
+            parsed_ual["contract_address"],
+            parsed_ual["token_id"],
+        )
+
+        self._reject_curated_node(
+            knowledge_asset_storage,
+            knowledge_asset_token_id,
+            identity_id,
+        )
+
+        return {
+            "paranetUAL": ual,
+            "paranetId": Web3.to_hex(
+                Web3.solidity_keccak(
+                    ["address", "uint256"],
+                    [knowledge_asset_storage, knowledge_asset_token_id],
+                )
+            ),
+            "operation": "Approve node's curated paranet access request"
+        }
+    
+    _get_curated_nodes = Method(BlockchainRequest.get_curated_nodes)
+    
+    def get_curated_nodes(self, ual: UAL) -> dict[str, str | HexStr | TxReceipt]:
+        parsed_ual = parse_ual(ual)
+        knowledge_asset_storage, knowledge_asset_token_id = (
+            parsed_ual["contract_address"],
+            parsed_ual["token_id"],
+        )
+
+        paranet_id = Web3.solidity_keccak(
+            ["address", "uint256"], [knowledge_asset_storage, knowledge_asset_token_id]
+        )
+
+        curated_nodes = self._get_curated_nodes(paranet_id)
+
+        # curated_nodes is a list of ABIDecodedNamedTuple type so it needs to be converted to a dictionary
```

Comment:
Can we perhaps make it smarter somehow?

Reply:
Not sure, I tried a couple solutions already. This is the simplest one I could come up with if we want to have a response look like in dkg.js:
```
{
    "0": 1,
    "1": "0x516d577966336474714a6e687543707a4544544e6d4e46596335746a7854725868476355556d474864673267746a",
    "identityId": 1,
    "nodeId": "0x516d577966336474714a6e687543707a4544544e6d4e46596335746a7854725868476355556d474864673267746a"
}
```

Reply:
We don't need it to be the same, this is not pythonic way